content/blog/host-based-volume-encryption-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,249 @@
+
+I conducted an experiment where I wrote an 8GiB file, filled with zeroes. Under normal circumstances, such a file is nearly invisible to any enterprise array. Not when encrypted.
+
+```markdown
+HPE Alletra $ vol --info  pvc-44159f94-0f2b-4109-ae90-e48d0df082b1 | grep ^Volume
+Volume mapped usage (MiB): 7859
+Volume compression: 0.97X
+```
+
+It's advised to consult with the storage administrator to ensure data-at-rest encryption on the array is turned off along with deduplication and compression to spare CPU cycles.
+
+Naturally, there will be a performance impact on the host and it's advised to study empirical data from benchmarks conducted with a production-like workload to understand the amount of CPU overhead is needed to ensure the application meets its performance criteria.
+
+# Maliciously trying to retrieve data
+
+In the event of a rogue host gain access to a volume without having the key, assume the LUN gets connected and discovered.
+
+```bash
+# mount /dev/mapper/mpatha /mnt
+mount: unknown filesystem type 'crypto_LUKS'
+```
+
+The encrypted volume may in this state be brute forced or maliciously deleted, it's advisable to use a strong non-dictionary passphrase to encrypt and decrypt with. The passphrase can be up to 512 characters long. The passphrase length does not affect performance or the cipher strength, it's only used to open the device to the host.
+
+As an extra layer of security when using the iSCSI protocol is to use the Challenge-Handshake Authentication Protocol (CHAP) facility available to the HPE CSI Driver, that ensures a mutual (between initiator and target) shared secret is needed to perform a discovery in the first place.
+
+# Summary
+
+Being able to confidently store sensitive data on storage systems out of your control over insecure networks is becoming more important in the era of data being the digital oil. With this solution, you can take your keys and walk away without any concern of your volumes (or replicas of them) being stolen, manipulated or sold to third parties.
+
+* Check out the release announcement of [HPE CSI Driver for Kubernetes 2.0.0](https://community.hpe.com/t5/Around-the-Storage-Block/HPE-CSI-Driver-for-Kubernetes-now-available-for-HPE-Alletra/ba-p/7136280)
+* Read the documentation on HPE Storage Container Orchestrator Documentation (SCOD) around the [host-based Volume Encryption](https://scod.hpedev.io/csi_driver/using.html#volume_encryption) feature
+* Learn about the [new multitenancy feature on HPE Alletra 6000 and Nimble Storage](https://scod.hpedev.io/container_storage_provider/hpe_alletra_6000/index.html#multitenant_deployment) to further improve security of your storage infrastructure
+
+The team hang out on Slack and eager to learn about your security challenges. Sign up at [slack.hpedev.io](http://slack.hpedev.io) and login to the community at [hpedev.slack.com](https://hpedev.slack.com), check out #kubernetes #nimblestorage and #3par-primera.
```

Comment:
Change "hang" to "hangs". Change "eager" to "is eager". Change "check out" to "to check out"